pkgs/top-level/haskell-packages.nix

```diff
@@ -161,11 +163,17 @@ in rec {
     };
     ghcjs = callPackage ../development/haskell-modules {
       ghc = compiler.ghcjs;
-      compilerConfig = callPackage ../development/haskell-modules/configuration-ghcjs.nix { };
+      compilerConfig =
+        extendCompose
+          (callPackage ../development/haskell-modules/configuration-ghcjs.nix { })
+          (callPackage ../development/haskell-modules/configuration-ghc-7.10.x.nix { });
```

Comment:
Why don't you pass the ghc-7.10.x-related overrides as `packageSetConfig` here? That would rectify the need for function composition here.